hydra-cluster/test/Test/Hydra/Cluster/FaucetSpec.hs

```diff
@@ -22,3 +28,31 @@ spec =
               replicateConcurrently_ 10 $ do
                 vk <- generate genVerificationKey
                 seedFromFaucet_ node vk 1_000_000 Normal (contramap FromFaucet tracer)
+
+  describe "returnFundsToFaucet" $
+    it "seedFromFaucet and returnFundsToFaucet work together" $ do
+      showLogsOnFailure $ \tracer ->
+        withTempDir "end-to-end-cardano-node" $ \tmpDir ->
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
+            let faucetTracer = contramap FromFaucet tracer
+            actor <- generate $ elements [Alice, Bob, Carol]
+            (vk, _) <- keysFor actor
+            (faucetVk, _) <- keysFor Faucet
+            initialFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
+            seeded <- seedFromFaucet node vk 100_000_000 Normal faucetTracer
+            returnFundsToFaucet faucetTracer node actor
+            remaining <- queryUTxOFor networkId nodeSocket QueryTip vk
+            finalFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
+            foldMap txOutValue remaining `shouldBe` mempty
+
+            let seededUTxOLength = length (pairs seeded)
+            let remainingUTxOLength = length (pairs remaining)
+            -- check the faucet has one utxo extra in the end
+            seededUTxOLength `shouldBe` remainingUTxOLength + 1
+
+            let initialFaucetValue = selectAsset (foldMap txOutValue initialFaucetFunds) AdaAssetId
+            let finalFaucetValue = selectAsset (foldMap txOutValue finalFaucetFunds) AdaAssetId
+            let difference = initialFaucetValue - finalFaucetValue
+            -- difference between starting faucet amount and final one should
+            -- just be the amount of paid fees
+            difference `shouldSatisfy` (< 340_000)
```

Comment:
can we fetch the amount of paid fees and use it here instead of hardcoded value?

Reply:
I think that would maybe a bit too hard here or at least we would need to also return the `tx` from one of the functions in order to inspect the fees.